src/game/challenges/batteries_challenges.js

```diff
@@ -0,0 +1,734 @@
+      Legal, não?
+      Com isto, fazemos com que ele percorra esse trajeto muito longo
+      com pouco código. Experimente você mesmo!
+    </p>
+  </div>,
+  commandReference,
+  "",
+  8,
+  10,
+  10,
+  new Grid.Position(0, 0),
+  Grid.Directions.RIGHT,
+  [
+    new Grid.Position(0, 2), new Grid.Position(0, 8),
+    new Grid.Position(0, 9), new Grid.Position(4, 9),
+    new Grid.Position(9, 9), new Grid.Position(9, 7),
+    new Grid.Position(9, 0), new Grid.Position(7, 0),
+    new Grid.Position(3, 0), new Grid.Position(2, 0)
+  ],
+  [
+    new Grid.Position(1, 5), new Grid.Position(2, 8),
+    new Grid.Position(8, 8), new Grid.Position(3, 7),
+    new Grid.Position(7, 4), new Grid.Position(1, 6)
+  ]
+);
+
+createBatteriesChallenge(
+  "step17",
+  <p>
+    Nosso último robô de hoje!
+    Mais dois laços aninhados e terminamos a missão!
+  </p>,
+  <div>
+    <p>
+      Muito bem!
+      Quando um laço está dentro de outro,
+      dizemos que eles estão aninhados.
+      Mas não se preocupe, é só o nome.
+    </p>
+    <p>
+      Este é nosso último robô precisando de ajuda hoje.
+      Com certeza vai ficar muito feliz com tantas baterias por aí!
+      Hmm, como são tantas, talvez seja mais fácil fazer o robô passar por
+      todos os lugares possíveis.
+      Ou melhor, quase todos: cuidado com as baterias ruins.
+      Você pode fazer isso com laços aninhados :)
+    </p>
+  </div>,
+  commandReference,
+  "",
+  20,
+  10,
+  10,
+  new Grid.Position(9, 9),
+  Grid.Directions.UP,
+  [
+    new Grid.Position(9, 3), new Grid.Position(9, 4),
+    new Grid.Position(9, 5), new Grid.Position(9, 8),
+    new Grid.Position(8, 2), new Grid.Position(8, 6),
+    new Grid.Position(8, 5), new Grid.Position(8, 3),
+    new Grid.Position(8, 7), new Grid.Position(8, 8),
+    new Grid.Position(7, 4), new Grid.Position(7, 5),
+    new Grid.Position(7, 8), new Grid.Position(7, 6),
+    new Grid.Position(6, 3), new Grid.Position(6, 9),
+    new Grid.Position(6, 5), new Grid.Position(5, 4),
+    new Grid.Position(5, 9), new Grid.Position(5, 8),
+    new Grid.Position(4, 5), new Grid.Position(4, 8),
+    new Grid.Position(4, 4), new Grid.Position(4, 6),
+    new Grid.Position(3, 2), new Grid.Position(3, 9),
+    new Grid.Position(3, 5), new Grid.Position(3, 8),
+    new Grid.Position(2, 2), new Grid.Position(2, 6),
+    new Grid.Position(1, 2), new Grid.Position(1, 7),
+    new Grid.Position(1, 9), new Grid.Position(1, 5),
+    new Grid.Position(0, 2), new Grid.Position(0, 3),
+    new Grid.Position(0, 4), new Grid.Position(0, 5),
+    new Grid.Position(0, 6), new Grid.Position(0, 7),
+    new Grid.Position(0, 9), new Grid.Position(0, 8)
+  ],
+  [
+    new Grid.Position(0, 0), new Grid.Position(0, 1),
+    new Grid.Position(2, 0), new Grid.Position(3, 1),
+    new Grid.Position(3, 0), new Grid.Position(8, 1),
+    new Grid.Position(9, 0), new Grid.Position(2, 1)
+  ]
+);
+
+createBatteriesChallenge(
```

Comment:
Can you remove this step? It was only meant to keep the students busy after the lesson, but now that they can go back and move on, it's useless. It's also (unintentionally) impossible, since there are green batteries underneath red ones :)